api/go.mod

```diff
@@ -1,72 +1,74 @@
 module github.com/openstack-k8s-operators/openstack-baremetal-operator/api
 
-go 1.18
+go 1.21
+
+toolchain go1.21.4
 
 require (
-	github.com/go-logr/logr v1.3.0
+	github.com/go-logr/logr v1.4.1
 	github.com/metal3-io/baremetal-operator/apis v0.3.1
 	github.com/onsi/ginkgo/v2 v2.14.0
 	github.com/onsi/gomega v1.30.0
 	github.com/openstack-k8s-operators/lib-common/modules/common v0.1.0
-	k8s.io/api v0.26.13
-	k8s.io/apimachinery v0.26.13
-	sigs.k8s.io/controller-runtime v0.14.7
+	k8s.io/api v0.29.1
```

Comment:
As commented on slack, we should not bump to a version beyond what the OCP version we're testing with supports.